components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -1584,28 +1531,23 @@ void AdsServiceImpl::ShowNotificationAd(base::Value::Dict dict) {
     notification->set_never_timeout(true);
 #endif
 
-    display_service_->Display(NotificationHandler::Type::BRAVE_ADS,
-                              *notification, /*metadata=*/nullptr);
+    delegate_->Display(*notification);
   }
 
   StartNotificationAdTimeOutTimer(ad.placement_id);
 }
 
 void AdsServiceImpl::CloseNotificationAd(const std::string& placement_id) {
```

Comment:
Maybe we can move this function to `AdsServiceDelegate` and pass `ShouldShowCustomNotificationAds()` value as a parameter? Declaration can look something like:
```
virtual void CloseNotificationAd(
                                    bool should_show_custom_notification_ads,
                                    const std::string& placement_id);
```

If so, the new function will replace following functions in `AdsService::Delegate`:
```
    virtual void Close(const std::string& notification_id) = 0;
    virtual void CloseNotificationAd(const std::string& id) = 0;
```